cmd/notation/internal/truststore/truststore.go

```diff
@@ -0,0 +1,226 @@
+// 'notation cert show' command
+var KeyUsageNameMap = map[x509.KeyUsage]string{
+	x509.KeyUsageDigitalSignature:  "Digital Signature",
+	x509.KeyUsageContentCommitment: "Non Repudiation",
+	x509.KeyUsageKeyEncipherment:   "Key Encipherment",
+	x509.KeyUsageDataEncipherment:  "Data Encipherment",
+	x509.KeyUsageKeyAgreement:      "Key Agreement",
+	x509.KeyUsageCertSign:          "Certificate Sign",
+	x509.KeyUsageCRLSign:           "CRL Sign",
+	x509.KeyUsageEncipherOnly:      "Encipher Only",
+	x509.KeyUsageDecipherOnly:      "Decipher Only",
+}
+
+// ExtKeyUsagesNameMap is a map of x509.Certificate ExtKeyUsages map used
+// in 'notation cert show' command
+var ExtKeyUsagesNameMap = map[x509.ExtKeyUsage]string{
+	x509.ExtKeyUsageAny:                            "Any",
+	x509.ExtKeyUsageServerAuth:                     "TLS Web Server Authentication",
+	x509.ExtKeyUsageClientAuth:                     "TLS Web Client Authentication",
+	x509.ExtKeyUsageCodeSigning:                    "Code Signing",
+	x509.ExtKeyUsageEmailProtection:                "E-mail Protection",
+	x509.ExtKeyUsageIPSECEndSystem:                 "IPSec End System",
+	x509.ExtKeyUsageIPSECTunnel:                    "IPSec Tunnel",
+	x509.ExtKeyUsageIPSECUser:                      "IPSec User",
+	x509.ExtKeyUsageTimeStamping:                   "Time Stamping",
+	x509.ExtKeyUsageOCSPSigning:                    "OCSP Signing",
+	x509.ExtKeyUsageMicrosoftServerGatedCrypto:     "Microsoft Server Gated Crypto",
+	x509.ExtKeyUsageNetscapeServerGatedCrypto:      "Netscape Server Gated Crypto",
+	x509.ExtKeyUsageMicrosoftCommercialCodeSigning: "Microsoft Commercial Code Signing",
+	x509.ExtKeyUsageMicrosoftKernelCodeSigning:     "Microsoft Kernel Code Signing",
+}
+
+// AddCert adds a single cert file at path to the trust store
+// under dir truststore/x509/storeType/namedStore
+func AddCert(path, storeType, namedStore string, display bool) error {
+	// initialize
+	certPath, err := filepath.Abs(path)
+	if err != nil {
+		return err
+	}
+	if storeType == "" {
+		return errors.New("store type cannot be empty")
+	}
+	if !ValidateStoreType(storeType) {
+		return fmt.Errorf("unsupported store type: %s", storeType)
+	}
+	if namedStore == "" {
+		return errors.New("named store cannot be empty")
+	}
+
+	// check if the target path is a cert (support PEM and DER formats)
+	if _, err := corex509.ReadCertificateFile(certPath); err != nil {
+		return err
+	}
+
+	// core process
+	// get the trust store path
+	trustStorePath, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err := CheckNonErrNotExistError(err); err != nil {
+		return err
+	}
+	// check if certificate already in the trust store
+	if _, err := os.Stat(filepath.Join(trustStorePath, filepath.Base(certPath))); err == nil {
+		return errors.New("certificate already exists in the Trust Store")
+	}
+	// add cert to trust store
+	_, err = osutil.CopyToDir(certPath, trustStorePath)
```

Comment:
We should never override existing cert unless explicitly specified by user.

Reply:
That's correct. Note, code right above this logic checks existence first. If a cert fileName is already in the specified truststore/x509/storeType/namedStore, `return errors.New("certificate already exists in the Trust Store")`.